Fixes HE Pipes

Zero-Bits opened the pull request:
:cl: ZeroBits
bugfix: "Heat Exchange Pipes can now be properly constructed."
tweak: "Heat Exchange Junctions will now exchange heat."
/:cl:

Also removed some relative pathing in favor of absolute pathing.

<!-- 
Do not forget to add a changelog when you have made admin/player facing changes that can alter gameplay.
Examples which require a changelog entry include:
* Adding/removing objects that players may interact with, or the way they function.
* Adding/removing/altering admin tools.
* Changing the map.

Examples were changelog entries are optional/not typically required:
* Cosmetic changes such as descriptions, sound effects, etc.
* Optimizations and other changes to underlying systems which do not affect gameplay.
* Minor bug fixes.

You find a README and example file in .\html\changelogs\ for further instructions.
-->

Zero-Bits committed:
Fix HE Pipe Construction

Fixes

atlantiscze commented:
IIRC there is a heat exchange junction under one of the windows in the supermatter core (fuel heat exchanger), which means it will now exchange heat and heat up engine room when outer window is removed (people often do this briefly to stuff RCAs in there instead of grilles).

I would recommend adjusting it by moving the junction into the core.

Techhead0 commented on code/game/machinery/pipe/construction.dm:
```diff
@@ -777,7 +777,7 @@ Buildable meters
 			var/obj/machinery/atmospherics/pipe/simple/heat_exchanging/junction/P = new ( src.loc )
 			P.set_dir(src.dir)
-			P.initialize_directions = src.get_pdir()
+			P.initialize_directions = src.get_pdir() + src.get_hdir()
```
Use | here instead of +. If pdir() and hdir() happen to overlap, + is gonna create some weird hard-to-diagnose behavior.

Zero-Bits replied:
Ah, didn't know if | would work properly through functions/vars like that. I'll get on it.

Techhead0 replied:
Works as long as the procs continue to give the expected output and you don't end up with a type mismatch.

Techhead0 commented on code/modules/atmospherics/he_pipes.dm:
```diff
@@ -113,40 +110,41 @@ obj/machinery/atmospherics/pipe/simple/heat_exchanging/junction
+obj/machinery/atmospherics/pipe/simple/heat_exchanging/junction/atmos_init()
+	..()
+	// Only check back side for normal pipes
+	for(var/obj/machinery/atmospherics/target in get_step(src,turn(src.dir, 180)))
```
Instead of `turn()`, you can use `GLOB.flip_dir[src.dir]`.

Zero-Bits replied:
:+1:

Zero-Bits replied:
:+1:

Zero-Bits commented:
I'll check it out @atlanticze, if that is the case, I'll have it fixed.

Zero-Bits commented:
Hokay, fixed the engine pipe that @atlantiscze pointed out, and implemented suggestions from @Techhead0 

I also straightened out that cold-loop pipe that had the stupid bend in it, and set the phoron pump to be on by default, so Phoron no longer builds up in the filter if you forget.
